ydb/core/tx/datashard/ut_common/datashard_ut_common.cpp

```diff
@@ -34,6 +34,67 @@ using namespace Tests;
 const bool ENABLE_DATASHARD_LOG = true;
 const bool DUMP_RESULT = false;
 
+namespace {
+
+    TEvDataShard::TEvGetInfoResponse* GetEvGetInfo(
+        Tests::TServer::TPtr server,
+        ui64 tabletId,
+        TAutoPtr<IEventHandle>& handle)
+    {
+        auto &runtime = *server->GetRuntime();
+
+        auto sender = runtime.AllocateEdgeActor();
+        auto request = MakeHolder<TEvDataShard::TEvGetInfoRequest>();
+        runtime.SendToPipe(tabletId, sender, request.Release(), 0, GetPipeConfigWithRetries());
+
+        TTableInfoMap result;
+
+        return runtime.GrabEdgeEventRethrow<TEvDataShard::TEvGetInfoResponse>(handle);
+    }
+
+    void SendReadTablePart(
+        Tests::TServer::TPtr server,
+        ui64 tabletId,
+        const TTableId& tableId,
+        const NKikimrTxDataShard::TEvGetInfoResponse::TUserTable& userTable,
+        ui64 readId,
+        NKikimrDataEvents::EDataFormat format)
+    {
+        auto& runtime = *server->GetRuntime();
+        auto sender = runtime.AllocateEdgeActor();
+
+        auto request = GetBaseReadRequest(tableId, userTable, readId, format);
+
+        AddFullRangeQuery(*request);
+
+        SendReadAsync(server, tabletId, request.release(), sender);
+    }
+
+    void PrintTableFromResult(TStringBuilder& out, const TEvDataShard::TEvReadResult& event, const NKikimrTxDataShard::TEvGetInfoResponse::TUserTable& userTable) {
+        const auto& description = userTable.GetDescription();
```

Comment:
Only `description` is needed in this function. So, it can passed to the function and `userTable` could be deleted.

Reply:
From any shard?

Reply:
You should call `userTable.GetDescription()` outside `PrintTableFromResult` and pass only `description` inside.